addons/web/static/src/search/search_bar/search_bar.js

```diff
@@ -98,17 +108,39 @@ export class SearchBar extends Component {
             return;
         }
 
-        for (const searchItem of this.searchItems) {
+        // move properties under their related field
+        const searchItems = [...this.searchItems];
+        searchItems.sort((a, b) => (a.propertyItemId || a.id) - (b.propertyItemId || b.id));
+
+        for (const searchItem of searchItems) {
             const field = this.fields[searchItem.fieldName];
-            const type = field.type === "reference" ? "char" : field.type;
+            let type = field.type === "reference" ? "char" : field.type;
+
+            if (type === "properties" && searchItem.type === "field_property") {
```

Comment:
It's a bit confusing to have a type on the field of the searchItem AND a type on the searchItem itself.

Did searchItem have types before your changes?
If not, I would just add some dict key, like "isFieldPropertyItem" to true, and change the checks to that, would probably be easier to read.